src/lib/fhirValidator.ts

```diff
@@ -89,8 +37,127 @@ const fhirValidator = {
-      returnPromise = Promise.resolve(this.validator[resourceType]);
+      this.validator[version][resourceType] = resource => resource.id && resource.id.length;
+      returnPromise = Promise.resolve(this.validator[version][resourceType]);
+    } else {
+      // eslint-disable-next-line no-lonely-if
+      if (version === FHIR_VERSION_R4) {
+        if (resourceType === 'Encounter') {
+          returnPromise = import('../../fhir/r4/encounter').then(bundle => {
+            this.validator[version][resourceType] = bundle.default;
+            return this.validator[version][resourceType];
+          });
+        }
+        if (resourceType === 'DocumentReference') {
+          returnPromise = import('../../fhir/r4/documentreference').then(bundle => {
+            this.validator[version][resourceType] = bundle.default;
+            return this.validator[version][resourceType];
+          });
+        }
+
+        if (resourceType === 'Practitioner') {
+          returnPromise = import('../../fhir/r4/practitioner').then(bundle => {
+            this.validator[resourceType] = bundle.default;
+            return this.validator[resourceType];
+          });
+        }
+
+        if (resourceType === 'Patient') {
+          returnPromise = import('../../fhir/r4/patient').then(bundle => {
+            this.validator[version][resourceType] = bundle.default;
+            return this.validator[version][resourceType];
+          });
+        }
+
+        if (resourceType === 'Questionnaire') {
+          returnPromise = import('../../fhir/r4/questionnaire').then(bundle => {
+            this.validator[version][resourceType] = bundle.default;
+            return this.validator[version][resourceType];
+          });
+        }
+
+        if (resourceType === 'QuestionnaireResponse') {
+          returnPromise = import('../../fhir/r4/questionnaireresponse').then(bundle => {
+            this.validator[version][resourceType] = bundle.default;
+            return this.validator[version][resourceType];
+          });
+        }
+
+        if (resourceType === 'DiagnosticReport') {
+          returnPromise = import('../../fhir/r4/diagnosticreport').then(bundle => {
+            this.validator[version][resourceType] = bundle.default;
+            return this.validator[version][resourceType];
+          });
+        }
+        if (resourceType === 'ResearchSubject') {
+          returnPromise = import('../../fhir/r4/researchsubject').then(bundle => {
+            this.validator[version][resourceType] = bundle.default;
+            return this.validator[version][resourceType];
+          });
+        }
+        if (resourceType === 'Observation') {
+          returnPromise = import('../../fhir/r4/observation').then(bundle => {
+            this.validator[version][resourceType] = bundle.default;
+            return this.validator[version][resourceType];
+          });
+        }
+      } else {
```

Comment:
It's not actually possible to get to the validaatotr without R4 or STU3 being set as the resource type because the check for supported resources happens earlier so I thought I'd be a bit less verbose here; happy to add it though.